src/librustc_resolve/diagnostics.rs

```diff
@@ -964,6 +964,31 @@ impl<'a> Resolver<'a> {
 
         err.emit();
     }
+
+    crate fn make_undeclared_type_suggestion(
+        &mut self,
+        ident: Ident,
+    ) -> (String, Option<Suggestion>) {
+        let parent_scope = &ParentScope::module(self.graph_root);
```

Comment:
`parent_scope` is available inside `fn resolve_path_with_ribs`, no need to create a fake one.